bin/compute_gene_scores.R

```diff
@@ -0,0 +1,592 @@
+            is.character(methods), length(methods)>0)
+
+    scores <- lapply(methods, function(x) calculateSignatureScores(xp, genes, x))
+    names(scores) <- methods
+    return(scores)
+}
+
+
+
+## Check consistency across scores
+
+# add suffix to rownames
+reformatRownames <- function(x, n){
+    rownames(x) <- rownames(x) %>% str_replace_all("\\s|\\-",".") %>% paste0(".",n)
+    return(x)
+}
+
+
+# generic function to generate a correlation matrix, preferably based on output of multipleSignatureScores
+correlationMatrix <- function(score.list, names.list=NULL) {
+    stopifnot(is.list(score.list))
+
+    if(is.null(names.list)) names.list <- names(score.list)
+    stopifnot(length(score.list)==length(names.list))
+    renamed.score.list <- list()
+    for(i in 1:length(names.list)) {
+        renamed.score.list[[i]] <- reformatRownames(score.list[[i]], names.list[i])
+    }
+    names(renamed.score.list) <- names(score.list)
+
+    combined.matrix <- do.call(rbind, renamed.score.list)
+
+    cor.matrix.all <- cor(t(combined.matrix))
+
+    return(cor.matrix.all)
+}
+
+# legacy function
+correlationMatrixOld <- function(plageScores = NULL, gsvaScores = NULL, ssgseaScores = NULL, paiScores = NULL, singScores = NULL) {
+
+    if(!is.null(plageScores)) plageScores <- reformatRownames(plageScores,"PLAGE")
+    if(!is.null(gsvaScores)) gsvaScores  <- reformatRownames(gsvaScores, "GSVA")
+    if(!is.null(ssgseaScores)) ssgseaScores <- reformatRownames(ssgseaScores, "SSGSEA")
+    if(!is.null(paiScores)) paiScores <- reformatRownames(paiScores, "PAI")
+    if(!is.null(singScores))  singScores <- reformatRownames(singScores, "Singscore")
+
+    combined.matrix <- rbind(plageScores, gsvaScores, ssgseaScores, paiScores, singScores)
+
+    cor.matrix.all <- cor(t(combined.matrix))
+
+    return(cor.matrix.all)
+}
+
+
+## Comparisons between groups
+
+## Boxplots
+
+boxplotScores <- function(score.matrix, set, group, title="") {
+    bp.df <- data.frame(score=score.matrix[set,], group=group)
+    ggplot(bp.df, aes(x=group, y=score)) +
+        geom_boxplot() +
+        geom_point() +
+        ggtitle(title)
+}
+
+
+sessionInfo()
+
+
+####Commandline Argument parsing###
+args = commandArgs(trailingOnly=TRUE)
+if (length(args) < 3) {
+    stop("Usage: process_nanostring_data.R <scores.yaml> <counts.tsv> algorithm", call.=FALSE)
+}
+####Uncomment if debugging
+
+gene_sets_to_compute <- args[1]
+counts_input <- args[2]
+score.method <- args[3]
+
+#Input parsing yaml
+yaml_parsed <- read_yaml(gene_sets_to_compute)
+
+#Get GEX into shape for Nanostring pipeline
+counts <- read.table(counts_input, sep="\t", check.names=F, header=T) %>%
+    filter(`CodeClass` == "Endogenous") %>%
+    select(-`CodeClass`) %>%
+    column_to_rownames("Name")
+
+# explicit signature check against expression matrix to generate an QC overview
+cs <- checkSignatures(counts, yaml_parsed)
+
+#Write to multiQC output
+
+qc.file="signature_scores_qc_mqc.txt"
```

Comment:
I think we should distinguish here between the normalized and non-normalized counts otherwise only for one the results will end up in the MultiQC report.